tests/integration/test_main.py

```diff
@@ -0,0 +1,52 @@
+from fastapi.testclient import TestClient
+from unittest.mock import patch
+from main import app
+import json
+
+
+# Create a test client for the FastAPI app
+client = TestClient(app)
+
+
+def test_search_issues(s3_bucket, duckdb_connection):
+    # Prepare test params
+    params = {
+        "dataset": "conservation-area",
+        "offset": 0,
+        "limit": 10,
+    }
+
+    # Test the function that interacts with DuckDB and S3 via LocalStack
+    with patch("db.duckdb.connect", return_value=duckdb_connection):
+        response = client.get("/log/issue", params=params)
+
+    # Validate the results from the search
+    assert response.status_code == 200
+
+    response_data = response.json()
+    assert "X-Pagination-Total-Results" in response.headers
+    assert response.headers["X-Pagination-Total-Results"] == str(83)
+
+    assert len(response_data) > 0
+    assert response_data[0]["dataset"] == "conservation-area"
+    assert (
+        response_data[0]["resource"]
+        == "0b4284077da580a6daea59ee2227f9c7c55a9a45d57ef470d82418a4391ddf9a"
+    )
+
+
+def test_search_issues_no_parameters(duckdb_connection):
+    # Prepare test params
+    params = {}
+
+    # Test the function that interacts with DuckDB and S3 via LocalStack
+    with patch("db.duckdb.connect", return_value=duckdb_connection):
```

Comment:
Again, same as above - can we not let the integration test go right through the db package which will then use DuckDb itself to connect to the S3 bucket being provided by localstack?

Reply:
If we do not mock the duck db connection, it tries to access an actual S3 endpoint which do not exist. Only when i mock it and provide the s3_endpoint as testcontainer's url, it works.